components/brave_wallet/browser/eth_pending_tx_tracker.cc

```diff
@@ -28,15 +29,19 @@ EthPendingTxTracker::EthPendingTxTracker(EthTxStateManager* tx_state_manager,
       weak_factory_(this) {}
 EthPendingTxTracker::~EthPendingTxTracker() = default;
 
-bool EthPendingTxTracker::UpdatePendingTransactions(size_t* num_pending) {
+bool EthPendingTxTracker::UpdatePendingTransactions(
+    const absl::optional<std::string>& chain_id,
+    size_t* num_pending,
+    std::set<std::string>* pending_chain_ids) {
+  CHECK(num_pending && pending_chain_ids);
   base::Lock* nonce_lock = nonce_tracker_->GetLock();
   if (!nonce_lock->Try())
     return false;
```

Comment:
Btw what is the purpose of this lock? I can't find any threads around there now.

Reply:
It was there when we first wrote nonce tracker to be thread safe but we ended up have no multiple threads accessing it. We can remove it in a follow up PR.